addons/website_slides/views/website_slides_templates_lesson_fullscreen.xml

```diff
@@ -18,8 +18,25 @@
                     <a class="o_wslides_fs_toggle_sidebar d-flex align-items-center px-3" href="#" title="Lessons">
                         <i class="fa fa-bars"/><span class="d-none d-md-inline-block ml-1">Lessons</span>
                     </a>
-                    <a class="o_wslides_fs_review d-flex align-items-center px-3" t-att-href="slide.channel_id.website_url + '?active_tab=review'" title="Reviews" t-if="slide.channel_id.allow_comment">
-                        <i class="fa fa-pencil"/><span class="d-none d-md-inline-block ml-1">Write a review</span>
+                    <a class="o_wslides_fs_review d-flex align-items-center" title="Reviews" t-if="slide.channel_id.allow_comment">
+                        <t t-call="portal_rating.rating_stars_static_popup_composer">
+                            <t t-set="rating_avg" t-value="rating_avg"/>
+                            <t t-set="rating_total" t-value="rating_count"/>
+                            <t t-set="object" t-value="channel"/>
+                            <t t-set="token" t-value="channel.access_token"/>
+                            <t t-set="hash" t-value="message_post_hash"/>
+                            <t t-set="pid" t-value="message_post_pid"/>
+                            <t t-set="default_message" t-value="last_message"/>
+                            <t t-set="default_message_id" t-value="last_message_id"/>
+                            <t t-set="default_rating_value" t-value="last_rating_value"/>
+                            <t t-set="default_attachment_ids" t-value="last_message_attachment_ids"/>
+                            <t t-set="force_submit_url" t-value="'/slides/mail/update_comment' if last_message_id else False"/>
```

Comment:
Isn't most of this duplicated?
Maybe we could make a small "website_slides.rating_stars_static_popup_composer" template that just calls the other template and set common variables to avoid duplication.

Reply:
@awa-odoo It's not a pure duplicated, there're some difference (CSS classes)

It's not strange to have 2 different templates for the same thing ? (the real one and a new one which set the variable ?)

Or maybe set the variables directly in the template `rating_stars_static_popup_composer` itself, but it's not super clean IMO

Reply:
I was thinking something like:

```xml
<!-- Small template to set common variables when using the composer in website_slides -->
<template id="rating_stars_static_popup_composer">
     <t t-set="rating_avg" t-value="rating_avg"/>
     <t t-set="object" t-value="channel"/>
     <t t-set="token" t-value="channel.access_token"/>
     <t t-set="hash" t-value="message_post_hash"/>
     <t t-set="pid" t-value="message_post_pid"/>
     <t t-set="default_message" t-value="last_message"/>
     <t t-set="default_message_id" t-value="last_message_id"/>
     <t t-set="default_rating_value" t-value="last_rating_value"/>
     <t t-set="default_attachment_ids" t-value="last_message_attachment_ids"/>
     <t t-set="force_submit_url" t-value="'/slides/mail/update_comment' if last_message_id else False"/>
     <t t-set="disable_composer" t-value="not channel.can_review"/>
</template>
```

Since you're using a common method to prepare values, it would make sense to use a common template that's using those prepared values, right?

Reply:
And then when using it:

```xml
<t t-call="website_slides.rating_stars_static_popup_composer">
    <t t-set="icon" t-value="'fa fa-pencil'"/>
    <t t-set="_link_btn_classes" t-value="'d-inline-block text-white font-weight-light shadow-none'"/>
    <t t-set="_text_classes" t-value="'d-none d-md-inline-block'"/>
    <t t-set="hide_rating_avg" t-value="True"/>
</t>
```

Reply:
Mhhh ok, I'm not a fan of this solution but ok

Reply:
@awa-odoo Or we could move the logic inside the template itself, so we do not have 2 different templates for the same purpose, we don't have code duplication,...

Something like this

```xml
    <template id="rating_stars_static_popup_composer" name="Rating: rating composer in popup">
        <t t-set="disable_composer" t-value="not channel.can_review"/>
        <t t-set="display_composer" t-value="not disable_composer and not request.session.is_website_user"/>

        <div class="d-print-none o_rating_popup_composer o_not_editable p-0"
            t-att-data-rating_avg="rating_avg or 0.0"
            t-att-data-rating_count="rating_count or 0.0"
            t-att-data-token="channel.access_token"
            t-att-data-hash="message_post_hash"
            t-att-data-pid="message_post_pid"
            t-att-data-res_model="channel._name"
            t-att-data-res_id="channel.id"
            t-att-data-partner_id="request.env.user.partner_id.id"
            t-att-data-default_message="last_message"
            t-att-data-default_message_id="last_message_id"
            t-att-data-default_rating_value="last_rating_value"
            t-att-data-default_attachment_ids="last_message_attachment_ids"
            t-att-data-force_submit_url="'/slides/mail/update_comment' if last_message_id else False"
            t-att-data-disable_composer="disable_composer"
            t-att-data-display_composer="display_composer"
            t-att-data-link_btn_classes="_link_btn_classes"
            t-att-data-icon="icon"
            t-att-data-text_classes="_text_classes"
            t-att-data-hide_rating_avg="hide_rating_avg">
            <div class="d-flex flex-wrap align-items-center">
                <div class="o_rating_popup_composer_stars text-nowrap"/>
                <button t-if="display_composer" type="button"
                    t-att-class="'btn ' + _link_btn_classes or 'btn-primary'"
                    data-toggle="modal" data-target="#ratingpopupcomposer">
                    <i t-if="icon" t-att-class="icon"/>
                    <span t-attf-class="#{_text_classes} o_rating_popup_composer_text">
                        <t t-if="default_message_id">Modify your review</t>
                        <t t-else="">Add a review</t>
                    </span>
                </button>
                <div class="o_rating_popup_composer_modal"/>
            </div>
        </div>
    </template>
```

What do you think ?

Reply:
(because otherwise we have one template which call another just to set the value, but the second is never call directly and it feel a little weird to me :/ )

Reply:
Well this is duplicating the code of the initial portal template so no, it's just moving the duplication somewhere else.
If you don't like my previous suggestion, just keep the code as it is then, no big deal anyway.